### DISCUSSION TITLE
Add custom data to call

### DISCUSSION THREAD

**Discussion by tdrz** [baresip/baresip#1901] [General]
Is it possible to add some __custom information__ when __placing a call__ such that I receive this info back when the __call state__ changes?
To be more specific, I would like to add a __custom id__ to the call and then monitor the call progress (call states) through either the console output or a module like __mqtt__.

**Comment by cspiel1** [comment]
Can't you use the call-id field?
E.g. module menu returns it as synchronous answer to command `dial`. Your application can then use the call-id to track the call. The ua-events contain this call-id. See https://github.com/baresip/baresip/blob/00c881ff66e61e126eefa4cbbbc2889bf8ff2675/modules/menu/static_menu.c#L499 !

**Reply by tdrz** [reply]
The calls that I am placing are initiated in different parts of the systems, which generate their own **call id**s. I would prefer to always use the id generated by the initiator of the call.
So for an **outbound** call, this would be the id generated by that part of the internal system that is initiating the call.
For an **inbound** call, it would be the **call_id** generated by baresip and will be propagated to the internal systems.

**Reply by cspiel1** [reply]
You might use a hash-map in your application to store application specific data and use the call-id as key.

**Reply by tdrz** [reply]
> You might use a hash-map in your application to store application specific data and use the call-id as key.

Definitely, it's just more state to be kept elsewhere. 

You could think of this custom information as a `void *obj` parameter in a typical C API that will be passed back whenever an event related to it occurs.

**Reply by cspiel1** [reply]
Your welcome to open a PR. You might add a setter function for `struct call`, and then in `event.c` function `event_encode_dict()` the user data has to be serialized. E.g. convert a pointer `void *obj` to a hex-string.

**Reply by tdrz** [reply]
Thanks! I had a look the other day and the biggest question for me was how to place the additional info in the actual command 
For example when using the console 
`/dial +1234567890 my_custominfo`

because I think there are some optional parameters that are expected by baresip already in that position.

Anyway, will try to figure it out.
